src/test/java/org/cactoos/http/HtWireTest.java

```diff
@@ -69,30 +76,71 @@ public void guessesCorrectPortForHttps() throws Exception {
         this.checkPorts("https://localhost", HtWireTest.HTTPS_PORT);
     }
 
+    @Test
+    public void guessesCorrectPortForExplicit() throws Exception {
+        final int port = 1234;
+        this.checkPorts(
+            new FormattedText("https://localhost:%d", port).asString(),
+            port
+        );
+    }
+
     @Test
     public void worksWithProvidedHostNameAndPort() throws IOException {
         new FtRemote(new TkText("Hello")).exec(
             home -> MatcherAssert.assertThat(
                 new TextOf(
                     new HtResponse(
                         new HtWire(home.getHost(), home.getPort()),
-                        new InputOf(
-                            new JoinedText(
-                                new TextOf("\r\n"),
-                                new TextOf("GET / HTTP/1.1"),
-                                new FormattedText(
-                                    "Host:%s",
-                                    home.getHost()
-                                )
-                            )
-                        )
+                        new GetInput(home)
                     )
                 ),
                 new TextHasString("HTTP/1.1 200")
             )
         );
     }
 
+    // @todo #63:30min HtWire should not close the socket opened to the remote
+    //  service (this includes not closing associated input and output
+    //  streams). It should instead return an `Input` with the socket's
+    //  inputstream "unread". Refactor accordingly and unignore this test.
+    @Ignore("see todo above")
+    @Test
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    public void closesSocketOnlyAfterResponseIsClosed() throws Exception {
+        new FtRemote(new TkText("Hey")).exec(
+            home -> {
+                @SuppressWarnings("resource")
+                final Socket socket = new Socket(
+                    home.getHost(),
+                    home.getPort()
+                );
+                try (InputStream ins = new HtWire(() -> socket).send(
+                    new GetInput(home)
+                ).stream()) {
+                    new Assertion<>(
+                        "must have a response",
+                        () -> ins.available(),
```

Comment:
@victornoel using `Inputstream#available()` is flaky due to the fact that it returns number of bytes available for the next read **without blocking**, ie. there may still be bytes available on the other side of the network connection.

The availability of the full inputstream could be tested with a fake object or something similar.

Reply:
@llorllale Indeed, I have simply replaced it with a call to read because all the `TextOf`, `BytesOf` and other way to read a stream in cactoos close the stream after the read it, which goes against the rest of the test :)